pkg/linode-bs/controllerserver.go

```diff
@@ -348,19 +348,20 @@ func (linodeCS *LinodeControllerServer) ListVolumes(ctx context.Context, req *cs
 	nextToken := ""
 
 	listOpts := linodego.NewListOptions(0, "")
-	if req.GetMaxEntries() > 0 {
-		if startingToken == "" {
-			listOpts.Page = 1
-		} else {
-			startingPage, errParse := strconv.ParseInt(startingToken, 10, 64)
-			if errParse != nil {
-				return nil, status.Error(codes.Aborted, fmt.Sprintf("Invalid starting token %v", startingToken))
-			}
-			listOpts.Page = int(startingPage)
+	if startingToken != "" {
+		startingPage, errParse := strconv.ParseInt(startingToken, 10, 64)
+		if errParse != nil {
+			return nil, status.Error(codes.Aborted, fmt.Sprintf("Invalid starting token %v", startingToken))
 		}
+
+		listOpts.Page = int(startingPage)
 		nextToken = strconv.Itoa(listOpts.Page + 1)
 	}
 
+	if req.GetMaxEntries() > 0 {
+		nextToken = strconv.Itoa(int(req.GetMaxEntries()))
```

Comment:
`nextToken` should always be set to a Linode page number, as written this would skip entire pages of volumes on subsequent invocations

This function is invoked by CSI multiple times: first with a "starting token" which takes a provider-specific pagination offset, then again with the "NextToken" returned by the previous invocation. (This scheme allows for both numerical paging and identity based paging that you might see with something like Kafka. Linode uses numerical paging.)

Reply:
I've confirmed your assessment. It appears the changes to this file around pagination is in response to csi-test being updated in the original PR, which tested things like respecting max entries. It appears to correctly react to the specified max entries, we must use it as page_size in our API, and then nextToken set in the lines above (to be the next page) would suffice.

However, linodego does not currently support specifying page_size